### ISSUE TITLE
Designer removes needed code

### ISSUE THREAD

**Issue by MisinformedDNA** [dotnet/winforms#5915]
* .NET Core Version: 
TargetFramework: net5.0-windows
SDKs: 6.0.100-rc.1.21463.6, 5.0.401

* Have you experienced this same bug with .NET Framework?: Haven't tried

**Problem description:**

When I try to edit a control from the designer. The designer will remove needed parts of the code from `*.Designer.cs` which results in run-time errors.

**Expected behavior:**

Designer doesn't remove necessary code.

**Minimal repro:**

1. Open control in designer
2. Edit a property

Designer removed the assignment of two fields resulting in NullReferenceExceptions at run-time.

[Video](https://user-images.githubusercontent.com/1784452/135894102-8e47abf4-5d00-4e2b-970a-21d8d9f236e3.mp4)

[Code](https://github.com/MisinformedDNA/winforms-designer-issue)

[Code after designer changes](https://github.com/MisinformedDNA/winforms-designer-issue/tree/actual)

NOTE: I inherited this code and have never used the designer on this control before, so maybe something the `designer.cs` or `resx` files are invalid in some way?

<details>
<summary>dotnet --info</summary>

> ❯ dotnet --info
> .NET SDK (reflecting any global.json):
>  Version:   
>  Commit:    e627d556a1
> 
> Runtime Environment:
>  OS Name:     Windows
>  OS Version:  10.0.18363
>  OS Platform: Windows
>  RID:         win10-x64
>  Base Path:   C:\Program Files\dotnet\sdk\6.0.100-rc.1.21463.6\
> 
> Host (useful for support):
>   Version: 6.0.0-rc.1.21451.13
>   Commit:  d7619cd4b1
> 
> .NET SDKs installed:
>   3.1.409 [C:\Program Files\dotnet\sdk]
>   5.0.400 [C:\Program Files\dotnet\sdk]
>   5.0.401 [C:\Program Files\dotnet\sdk]
>   6.0.100-rc.1.21463.6 [C:\Program Files\dotnet\sdk]
> 
> .NET runtimes installed:
>   Microsoft.AspNetCore.All 2.1.29 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
>   Microsoft.AspNetCore.App 2.1.29 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
>   Microsoft.AspNetCore.App 3.1.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
>   Microsoft.AspNetCore.App 3.1.18 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
>   Microsoft.AspNetCore.App 3.1.19 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
>   Microsoft.AspNetCore.App 5.0.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
>   Microsoft.AspNetCore.App 5.0.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
>   Microsoft.AspNetCore.App 6.0.0-rc.1.21452.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
>   Microsoft.NETCore.App 2.1.29 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
>   Microsoft.NETCore.App 3.1.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
>   Microsoft.NETCore.App 3.1.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
>   Microsoft.NETCore.App 3.1.18 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
>   Microsoft.NETCore.App 3.1.19 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
>   Microsoft.NETCore.App 5.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
>   Microsoft.NETCore.App 5.0.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
>   Microsoft.NETCore.App 6.0.0-rc.1.21451.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
>   Microsoft.WindowsDesktop.App 3.1.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
>   Microsoft.WindowsDesktop.App 3.1.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
>   Microsoft.WindowsDesktop.App 3.1.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
>   Microsoft.WindowsDesktop.App 3.1.18 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
>   Microsoft.WindowsDesktop.App 3.1.19 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
>   Microsoft.WindowsDesktop.App 5.0.9 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
>   Microsoft.WindowsDesktop.App 5.0.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
>   Microsoft.WindowsDesktop.App 6.0.0-rc.1.21451.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
> 

<details>

**Comment by Tanya-Solyanik** [comment]
DevExpress package used in the project probably does not have the Code designers.

**Comment by MisinformedDNA** [comment]
`DataSet` is getting unassigned as well. And there are other DevExpress controls that work fine.

**Comment by MisinformedDNA** [comment]
@Tanya-Solyanik  I didn't want to edit the designer file for my repro, but I went ahead and did it. DevExpress is now removed as a dependency. I updated [main](https://github.com/MisinformedDNA/winforms-designer-issue) and here's the new [change commit](https://github.com/MisinformedDNA/winforms-designer-issue/commit/f4a84b98fa151f05250a3df0155105edf598fbfb). Same issue.

**Comment by RussKie** [comment]
@MisinformedDNA have you contacted DevExpress?

**Comment by MisinformedDNA** [comment]
@RussKie I shared a repro with DevExpress removed. It's clearly not them.

**Comment by dreddy-work** [comment]
@KlausLoeffelmann , can you take a look at the sample here and see anything stand out?

**Comment by MisinformedDNA** [comment]
@KlausLoeffelmann @dreddy-work Can I get some help? How do I get this resolved?

**Comment by KlausLoeffelmann** [comment]
Hey @MisinformedDNA,

thanks for reporting that. We would need a repo which doesn't use a third party custom control, of which we don't know how compatible it is with the new designer.

Please also note that Designer support for Typed DataSets is limited for .NET WinForms Apps, since the Data Source Provider Service (which are necessary for maintaining Typed DataSets and get the necessary components into a Form/UserControl on demand) is not available for .NET apps.

That said, the most important scenarios we enabled (in WinForms! WPF is a different story, btw) _without_ resorting to the Data Source Provider Service, so, I'd be interested if we can find a purely WinForms based scenario for this, since this would _need_ to be fixed then!

**Comment by MisinformedDNA** [comment]
@KlausLoeffelmann The project does not depend on a 3rd party, that dependency was removed Oct 5 (see above).

The note about designer support for Typed DataSets is more of an FYI, since I'm using WinForms? Is that correct?

**Comment by KlausLoeffelmann** [comment]
> The note about designer support for Typed DataSets is more of an FYI, since I'm using WinForms? Is that correct?

No, there might be issues, but I never personally encountered code-loss with using them.
And sorry, I was watching your video primarily and saw it had the dependency to DevExpress. I missed that you removed that dependency. I would need to investigate some more, but it's taking a few days at least, so I'd ask to hang in there for a bit longer!

**Comment by MisinformedDNA** [comment]
@KlausLoeffelmann Thanks for looking into it!

**Comment by MisinformedDNA** [comment]
@KlausLoeffelmann Any update?

**Comment by RussKie** [comment]
@KlausLoeffelmann is currently OOF. With holiday season is almost upon us our responses may be delayed.

**Comment by KlausLoeffelmann** [comment]
That said, I know my dept, even watching my Mom-visiting-from-Germany backing Holiday cookies. Will take a while though, still... 🎅

**Comment by merriemcgaw** [comment]
@KlausLoeffelmann getting this back on your radar.

**Comment by KlausLoeffelmann** [comment]
Follow up question:
From your demo, I can't really see for component the generated code is swallowed.
Can you provide the sample without any references to third party controls, so I can debug into the repo?

Are you using a .NET Framework control by any chance in the context of a .NET (Core) application?

**Comment by MisinformedDNA** [comment]
You guys keep asking about a sample without third party controls, but the linked code in OP does not use any third party controls.
